pkg/util/log/eventpb/cluster_events.proto

```diff
@@ -57,6 +57,34 @@ message NodeRestart {
   CommonNodeEventDetails node = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "", (gogoproto.embed) = true];
 }
 
+// NodeShutdownConnectionTimeout is recorded when SQL connections remain open
+// during shutdown, after waiting for the server.shutdown.connections.timeout
+// to transpire.
+message NodeShutdownConnectionTimeout {
+  CommonEventDetails common = 1 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "", (gogoproto.embed) = true];
+  CommonNodeEventDetails node = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "", (gogoproto.embed) = true];
+  // The detailed message, meant to be a human-understandable explanation.
+  string detail = 3 [(gogoproto.jsontag) = ",omitempty", (gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/cockroachdb/redact.SafeString", (gogoproto.moretags) = "redact:\"nonsensitive\""];
+  // The number of connections still open after waiting for the client to close them.
+  uint64 connections_remaining = 4 [(gogoproto.jsontag) = ",includeempty"];
+  // The amount of time the server waited for the client to close the connections, defined by server.shutdown.connections.timeout.
+  uint64 timeout_millis = 5 [(gogoproto.jsontag) = ",includeempty"];
```

Comment:
any chance these uint64s can be uint32? we've observed a limitation when these events are encoded into JSON and parsed as binary64 format which has a lower upper bound on the integers it supports.

I think both of these counters should have plenty of room with a smaller int.

@kyle-a-wong what do you think?

Reply:
uint32 should be totally fine. the timeoutMS and connection count should never exceed more than a few hundred thousand, even in very extreme cases.